### PR TITLE
[Spring] Fix generated api ignore basePath

### LINKED ISSUE #1928: [Spring] Generated controllers ignore basePath
I have the following line in the openapi file:

```
basePath: "/v1"
```

I would expect generated controllers (interfaces) would use this basePath as prefix for all the paths, but they don't contain this "v1" and just use resource path:

```
@RequestMapping(value = "/abc", method = RequestMethod.GET)
```

I'm using latest stable release 3.3.4.

### PR DESCRIPTION
**Pull request by RomainPruvostMHH**
Fix #1928 

The base-path should be set on the Api interface. This way it is working with interfaceOnly. And it's compliant with Spring documentation : https://www.baeldung.com/spring-interface-driven-controllers

### Changes
- I removed it from the *apiController.mustache* file and added it to the *api.mustache* file
- I moved the *defaultBasePath.mustache* file in *src/main/resources/JavaSpring* so that it's common with all Spring server-side libraries (spring-boot, spring-mvc, spring-cloud (*see the notice below for spring-cloud*) ). The value of default context path is the value of contextPath variable.
- I added a unit test to verify it

### Refactoring
- I updated the outputDir variable to *samples/server/petstore/spring-stubs* in the  *bin/configs/spring-stubs.yaml* file because it generates Spring server-side classes. And not on the client-side as the output directory seemed to suggest. I deleted the old *samples/client/petstore/spring-stubs* directory.

### Notice
- Some files are updated in the samples/client/petstore/spring-cloud* directories. Actually, the spring-cloud library isn't consistent. Both server-side and client-side files are generated. That's why all generated xxxApi.java files in *samples/client/petstore/spring-cloud* directories are updated with this fix too.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@bbdouglas  (2017/07) @sreeshas  (2017/08) @jfiala  (2017/08) @lukoyanov  (2017/09) @cbornet  (2017/09) @jeff9finger  (2018/01) @karismann  (2019/03) @Zomzog  (2019/04) @lwlee2608  (2019/10) @nmuesch  (2021/01)

### CODE REVIEW

**Comment by RomainPruvostMHH** [comment]
@wing328 Could you solve the problem in CircleCI build. A lot of build are stuck caused by concurrency limitations. I don't know the cause. the first blocked build seems : https://app.circleci.com/pipelines/github/OpenAPITools/openapi-generator?branch=pull%2F10544  . But I don't know if this build is the root cause.

**Review by welshm — APPROVED** [review]

**Comment by welshm** [comment]
Looks like your samples might be out of date - rebase master and regenerate samples?

**Comment by BenjaminBusvel** [comment]
Hello, could you give me more informations when the issue can be merged in official release please ?

**Comment by RomainPruvostMHH** [comment]
@welshm the build finally passes

**Comment by wing328** [comment]
@RomainPruvostMHH thanks for the PR. Can you please PM me via Slack when you've time? Thanks.

https://join.slack.com/t/openapi-generator/shared_invite/zt-12jxxd7p2-XUeQM~4pzsU9x~eGLQqX2g

**Comment by RomainPruvostMHH** [comment]
Hi @wing328 , I'm back . This PR was synchronized with the master and can be merged.

File: samples/client/petstore/cpp-tiny/lib/service/AbstractService.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+// Go and comment out a certificate in root.cert, if you get an error here
+// Certificate from file
+const char* test_root_ca =
+#include "../../root.cert"
+;
+
+}; // end class
+}// namespace Tinyclient
+
+#endif /* TINY_CPP_CLIENT_ABSTRACTSERVICE_H_ */
```
Comment (wing328) [inline samples/client/petstore/cpp-tiny/lib/service/AbstractService.h:28]: Weird. There shouldn't be changed to the cp-ting client.

Comment (wing328) [reply samples/client/petstore/cpp-tiny/lib/service/AbstractService.h:28]: We'll merge this and clean up the cpp-tiny samples if needed.

File: modules/openapi-generator/src/main/resources/JavaSpring/api.mustache [hunk 2]
```diff
@@ -96,6 +96,9 @@ import javax.annotation.Generated;
 {{#virtualService}}
 @VirtualService
 {{/virtualService}}
+{{=<% %>=}}
```
Comment (MelleD) [inline modules/openapi-generator/src/main/resources/JavaSpring/api.mustache:99]: This should just be added if NO feign is used.

**Comment by YevgenyDrozdov** [comment]
I suppose, this PR might roll back the fix of this issue https://github.com/OpenAPITools/openapi-generator/issues/4525 which was delivered by https://github.com/OpenAPITools/openapi-generator/pull/12250

**Comment by eugenealeinikov** [comment]
@RomainPruvostMHH hi. Not so long ago I added the possibility to skip url param generation in feign client using 'useFeignClientUrl' property (https://github.com/OpenAPITools/openapi-generator/pull/12250). Now I see that you deleted this code in 
modules/openapi-generator/src/main/resources/JavaSpring/libraries/spring-cloud/apiClient.mustache. Could you please explain why?

**Pull request merged**